### PR TITLE
[WIP] Delete implicit constructors from Storage

### PR DESCRIPTION
**Pull request by ezyang**
Stack:
&nbsp;&nbsp;&nbsp;&nbsp;:black_circle:&nbsp; **#11700 Delete implicit constructors from Storage**&nbsp;&nbsp;[:green_heart:](https://our.internmc.facebook.com/intern/diff/D9830460/)

There isn't really any reason to ever have an implicit
conversion to Storage

Differential Revision: [D9830460](https://our.internmc.facebook.com/intern/diff/D9830460/)

### CODE REVIEW

File: caffe2/core/tensor.h [hunk 1]
```diff
@@ -53,6 +53,9 @@ class CAFFE2_API Tensor final {
   explicit Tensor(Storage storage)
       : impl_(c10::make_intrusive<TensorImpl, UndefinedTensorImpl>(std::move(storage))) {}
 
+  explicit Tensor(DeviceType device_type)
```
Comment (gchanan) [inline caffe2/core/tensor.h]: this is necessary?

**Review by gchanan — APPROVED** [review]

File: caffe2/core/tensor.h (hunk 1 above)
Comment (ezyang) [reply caffe2/core/tensor.h]: Yes, I accidentally deleted it in a previous PR (but it still worked because DeviceType implicitly converted into Storage).

**Comment by SsnL** [comment]
@ezyang Rebase and go?

**Comment by ezyang** [comment]
Rebase was botched, this needs some fixing.

**Comment by zdevito** [comment]
I am closing this because it shows up on my reviews requested list, which I am cleaning up, but I cannot remove myself as a reviewer because the reviews list is bugged out and lists me twice. I this is something you still need, feel free to re-open.

**Pull request closed without merging**